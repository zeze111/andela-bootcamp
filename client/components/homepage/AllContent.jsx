import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ReactTooltip from 'react-tooltip'

const AllContent = ({
  recipe: {
    id, name, prepTime, image
  }
}) => {
  const show = `${name.substring(0, 10)}...`;
  return (
    <li >
      <div className="col s2 offset-1">
        <div className="card hoverable grey lighten-4 home-cards">
          <div className="card-image">
            <img
              src={image || '/images/noimg.png'}
              alt="recipe"
              className="responsive-img image-style"
            />
            <div className="card-action">
              <Link
                to={`/recipe/${id}`}
                href={`/recipe/${id}`}
                className="home-text-style"
                data-tip={name}
              > {show} : {prepTime}
              <ReactTooltip/>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </li>

  );
};

AllContent.propTypes = {
  recipe: PropTypes.objectOf(PropTypes.any).isRequired
};

export default AllContent;
