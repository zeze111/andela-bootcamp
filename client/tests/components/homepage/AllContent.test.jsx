import React from 'react';

import AllContent from '../../../components/homepage/AllContent';
import store from '../../../store';

const props = {
  recipe: {
    id: 1,
    name: 'some name'
  }
};

describe('Test for All Content', () => {
  it('should render component', () => {
    const wrapper = shallow(<AllContent {...props} store={store}/> );
    expect(wrapper).toMatchSnapshot()
  });
});
