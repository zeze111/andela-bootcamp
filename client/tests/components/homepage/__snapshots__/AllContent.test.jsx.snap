// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for All Content should render component 1`] = `
<li>
  <div
    className="col s2 offset-1"
  >
    <div
      className="card hoverable grey lighten-4 home-cards"
    >
      <div
        className="card-image"
      >
        <img
          alt="recipe"
          className="responsive-img image-style"
          src="/images/noimg.png"
        />
        <div
          className="card-action"
        >
          <Link
            className="home-text-style"
            data-tip="some name"
            href="/recipe/1"
            replace={false}
            to="/recipe/1"
          >
             
            some name...
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            />
          </Link>
          <span
            className="home-text2"
          >
            <i
              className="material-icons home-icon2"
            >
              visibility
            </i>
          </span>
        </div>
      </div>
    </div>
  </div>
</li>
`;
